File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        start = 0
        end = len(nums) - 1

        lookup = set()

        for i in range(len(nums) - 2):
            target =  - nums[i]
            lookup.clear()
            for j in range(i+1, len(nums), 1):
                if target - nums[j] in lookup:
                    temp = [-target, target - nums[j], nums[j]]
                    temp.sort()
                    ans.append(temp)
                else:
                    lookup.add(nums[j])    
        return [list(a) for a in set(tuple(x) for x in ans)]            



        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans = new int[nums.length];
        int res = 1;
        int numOfZeros = 0;
        for(int num: nums){
            if(num == 0)
                numOfZeros++;
            else
                res *= num;    
        }
            
        if(numOfZeros == 0){
            for(int i=0; i < nums.length; i++){
                ans[i] = res/nums[i];
            }
        }else if(numOfZeros == 1){
            for(int i=0; i < nums.length; i++){
                if(nums[i] != 0)
                    ans[i] = 0;
                else
                    ans[i] = res;    
            }
        }else{
            for(int i=0; i < nums.length; i++){
                ans[i] = 0;    
            }
        }
        return ans;    
    }
}  
